pkg/kubelet/cri/remote/utils_test.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func TestVerifySandboxStatus(t *testing.T) {
+	validStatus := getValidPodSandboxStatus()
+	statusWithoutID := getValidPodSandboxStatus()
+	statusWithoutID.Id = ""
+	statusWithoutMetadata := getValidPodSandboxStatus()
+	statusWithoutMetadata.Metadata = nil
+	statusWithoutCreatedAt := getValidPodSandboxStatus()
+	statusWithoutCreatedAt.CreatedAt = 0
+	statusWithoutName := getValidPodSandboxStatus()
+	statusWithoutName.Metadata.Name = ""
+
+	testcases := []struct {
+		description      string
+		podSandboxStatus *runtimeapi.PodSandboxStatus
+		expectedResult   error
+	}{{
+		description:      "verifySandboxStatus succeeds",
+		podSandboxStatus: validStatus,
+		expectedResult:   nil,
+	}, {
+		description:      "verifySandboxStatus fails due to lack of Id",
+		podSandboxStatus: statusWithoutID,
+		expectedResult:   fmt.Errorf("Id is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of metadata",
+		podSandboxStatus: statusWithoutMetadata,
+		expectedResult:   fmt.Errorf("Metadata is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of CreatedAt",
+		podSandboxStatus: statusWithoutCreatedAt,
+		expectedResult:   fmt.Errorf("CreatedAt is not set"),
+	}, {
+		description:      "verifySandboxStatus fails due to lack of name",
+		podSandboxStatus: statusWithoutName,
+		expectedResult:   fmt.Errorf("Name, Namespace or Uid is not in metadata \"&PodSandboxMetadata{Name:,Uid:100,Namespace:foo-ns,Attempt:10,}\""),
+	}}
+
+	for _, tc := range testcases {
+		t.Logf("TestCase: %q", tc.description)
+		result := verifySandboxStatus(tc.podSandboxStatus)
+		assert.Equal(t, result, tc.expectedResult)
```

Comment:
```suggestion
		assert.Equal(t, tc.expectedResult, result)
```

nit: Testify expects the expected result to be provided before the actual result for `Equal` - Which makes it slightly easier to read output. https://godoc.org/github.com/stretchr/testify/assert#Equal

Reply:
Thanks, a nice point.
I will update here.